cognite/client/_cognite_client.py

```diff
@@ -36,6 +36,16 @@ class CogniteClient:
         timeout (int): Timeout on requests sent to the api. Defaults to 30 seconds.
         token (Union[str, Callable[[], str]]): A jwt or method which takes no arguments and returns a jwt to use for authentication.
             This will override any api-key set.
+        token_url (str): Optional url to use for token generation.
+            This will override the COGNITE_TOKEN_URL environment variable and only be used if both api-key and token are not set.
+        token_client_id (str): Optional client id to use for token generation.
+            This will override the COGNITE_CLIENT_ID environment variable and only be used if both api-key and token are not set.
+        token_client_secret (str): Optional client secret to use for token generation.
+            This will override the COGNITE_CLIENT_SECRET environment variable and only be used if both api-key and token are not set.
+        token_scopes (list): Optional list of scopes to use for token generation.
+            This will override the COGNITE_TOKEN_SCOPES environment variable and only be used if both api-key and token are not set.
+        token_custom_args (Dict): Additional arguments to use for token generation.
```

Comment:
I would note that these are passed to the OAuth library fetch_token function or something